aiidalab_widgets_base/elns.py

```diff
@@ -0,0 +1,285 @@
+import json
+from pathlib import Path
+
+import ipywidgets as ipw
+import requests_cache
+import traitlets
+from aiida.orm import Node, QueryBuilder
+from aiidalab_eln import get_eln_connector
+from IPython.display import clear_output, display
+
+ELN_CONFIG = Path.home() / ".aiidalab" / "aiidalab-eln-config.json"
+ELN_CONFIG.parent.mkdir(
+    parents=True, exist_ok=True
+)  # making sure that the folder exists.
+
+
+def connect_to_eln(eln_instance=None, **kwargs):
+    try:
+        with open(ELN_CONFIG, "r") as file:
+            config = json.load(file)
+    except (FileNotFoundError, json.JSONDecodeError, KeyError):
+        return None
```

Comment:
Not sure how else this could fail, but do we want to propagate a more detailed exception to the user instead of returning?